doi-transformer/src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java

```diff
@@ -549,7 +575,7 @@ private FileSet createFilseSet() {
     private URL extractFulltextLinkAsURL(CrossRefDocument document) {
         try {
             final Optional<URI> optionalURI = extractFirstLinkToSourceDocument(document);
-            return  optionalURI.isPresent() ? optionalURI.get().toURL() : null;
+            return optionalURI.isPresent() ? optionalURI.get().toURL() : null;
```

Comment:
Suggestion for the following two methods

```suggestion

 private URL extractFulltextLinkAsURL(CrossRefDocument document) {
        return extractFirstLinkToSourceDocument(document)
                   .map(this::transformToUrl)
                   .orElse(null);
    }

    private URL transformToUrl(URI uri) {
        return attempt(uri::toURL).orElse(fail -> handleMalformedUrlException());
    }

    private URL handleMalformedUrlException() {
        logger.warn("Malformed URL in CrossRef document");
        return null;
    }

    private URI extractFulltextLinkAsURI(CrossRefDocument document) {
        return extractFirstLinkToSourceDocument(document).orElse(null);
    }

```

Reply:
Nice! Done!